### PR TITLE
Merge of `wolk_tracker_features`.

### PR DESCRIPTION
**Pull request by nanavuletic**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -36,6 +36,16 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}    -Wall -Wextra -pedantic
                                             -Wvariadic-macros                                                          \
                                             -Wwrite-strings -Werror=non-virtual-dtor -Werror=return-type")
 
+
+
```
Comment (djesic) [inline CMakeLists.txt:40]: This is great!

File: CMakeLists.txt [hunk 2]
```diff
@@ -79,35 +88,37 @@ target_include_directories(${PROJECT_NAME}Tests SYSTEM PRIVATE ${CMAKE_LIBRARY_I
 set_target_properties(${PROJECT_NAME}Tests PROPERTIES INSTALL_RPATH "$ORIGIN/lib")
 add_dependencies(${PROJECT_NAME}Tests libgtest)
 
-add_custom_target(tests ALL ${PROJECT_NAME}Tests)
+add_custom_target(tests ${PROJECT_NAME}Tests)
```
Comment (djesic) [inline CMakeLists.txt:91]: And no more test builds also?

File: RELEASE_NOTES.txt [hunk 3]
```diff
@@ -5,6 +5,16 @@
 
 WolkGateway bridges communication between WolkAbout IoT platform and multiple devices connected to it.
 
+**Version 4.3.0**
+    - [IMPROVEMENT] - Created WolkDefault and WolkExternal that provide different interfaces for custom subdevice data providing.
```
Comment (djesic) [inline RELEASE_NOTES.txt:9]: child device

File: debian/changelog [hunk 4]
```diff
@@ -1,7 +1,12 @@
-wolkgateway (4.2.9) stable; urgency=medium
+wolkgateway (4.3.0) stable; urgency=medium
 
-  * Fixed the gateway update to allow us to pass if it has already been updated.
-  * Disabled the gateway update fow now, as it doesn't really have a purpose right now.
-  * Imported newest WolkSDK-Cpp that allows for optional `description` properties in JsonDto messages.
+  * Created WolkDefault and WolkExternal that provide different interfaces for custom subdevice data providing.
```
Comment (djesic) [inline debian/changelog:3]: child device

File: src/WolkBuilder.h [hunk 5]
```diff
@@ -148,6 +157,39 @@ class WolkBuilder final
...
+    /**
+     * @brief withExternalDataProvider Use external data provider instead of expecting data on local mqtt
+     * with WolkAbout protocol
+     * @param provider Implementation of DataProvider
+     * @return Reference to current wolkabout::WolkBuilder instance (Provides fluent interface)
+     */
+    WolkBuilder& withExternalDataProvider(DataProvider* provider);
+
+    /**
+     * @brief withProtocol Use external protocol implementatoion for data and status messages
```
Comment (djesic) [inline src/WolkBuilder.h:178]: typo

**Review by djesic — COMMENTED** [review]
I'll leave these comments so far, I viewed 23/95 files (up to `src/WolkExternal.cpp`)

**Review by milanzlicic — APPROVED** [review]

**Pull request merged**